### PR TITLE
split user solutions

### PR DESCRIPTION
**Pull request by DmytroOrlov**

### CODE REVIEW

File: server/app/views/addTask.scala.html [hunk 1]
```diff
@@ -3,17 +3,18 @@
...
-        @textarea(task(referenceSolution), 'rows -> "13", 'cols -> "80")
-        @textarea(task(suite), 'rows -> "13", 'cols -> "80")
-        <div class="submit">
-            <input type="submit" value="@msg("addTask")"/>
+    @main(msg("addTask")) {
+        <div><a href="@routes.Application.index()">@msg("home")</a></div>
+        <div class="solutionForm">
+        @form(routes.NewTask.postNewTask()) {
+            @textarea(task(taskDescription), 'rows -> "8", 'cols -> "80")
+            @textarea(task(solutionHeader), 'rows -> "3", 'cols -> "80")
```
Comment (novakov-alexey) [inline server/app/views/addTask.scala.html]: Can you show examples of header, body and footer? It is not clear how to use this feature from user perspective.

Comment (DmytroOrlov) [inline server/app/views/addTask.scala.html]: //header
class SubArrayWithMaxSum extends SubArrayWithMaxSumSolution {
  def apply(a: Array[Int]): Array[Int] = {
//body
    var currentSum = 0
    var maxSum = 0
    var left, right = 0
    var maxI = 0 //used when all negatives in the array

    for (i <- a.indices) {
      val incSum = currentSum + a(i)

      if (incSum > 0) {
        currentSum = incSum

        if (currentSum > maxSum) {
          maxSum = currentSum
          right = i
        }
      } else {
        left = i + 1
        right = left
        currentSum = 0
        if (a(i) > a(maxI)) maxI = i
      }
    }

    if (left == a.length) a.slice(maxI, maxI + 1)
    else a.slice(left, right + 1)
//footer
  }
}

Comment (novakov-alexey) [inline server/app/views/addTask.scala.html]: Understood now. The idea is quite interesting itself. What if user adds a test with several methods to implement?

Comment (DmytroOrlov) [inline server/app/views/addTask.scala.html]: Need another solution type :)

Comment (novakov-alexey) [inline server/app/views/addTask.scala.html]: I am not sure we need to merge this into master as it limits our functionality from user perspective.

**Pull request closed without merging**